current_fall2025/scripts/RAG.py

```
@@ -0,0 +1,246 @@
+
+    sql = f"""
+        SELECT 
+            document_id,
+            chunk_index,
+            chunk_text,
+            metadata,
+            1 - (embedding <=> {qvec_str}::vector) AS score
+        FROM gold.bpl_embeddings
+        ORDER BY embedding <=> {qvec_str}::vector
+        LIMIT %s;
+    """
```

Ruff (0.14.0)

Possible SQL injection vector through string-based query construction

(S608)